### PR TITLE
feat(parser): #1262 phase 2.1a structured CST for 14 single-line directives

### PR DESCRIPTION
**Pull request by robcohen**
## Summary

Phase 2.1a of #1262: structured CST emission for 14 single-line directive types.

> **Scope split:** the original PR 2.1 covered 15 directive kinds including TRANSACTION. Split into **PR 2.1a (this PR)** for the 14 single-line directives and **PR 2.1b** (next) for TRANSACTION header. TRANSACTION couples naturally with PR 2.2's posting/amount/cost-spec body parsing, so the split keeps each PR mechanically uniform and reviewable.

## What this PR adds

- **14 specific directive `SyntaxKind` variants:** `OPEN_DIRECTIVE`, `CLOSE_DIRECTIVE`, `BALANCE_DIRECTIVE`, `PAD_DIRECTIVE`, `EVENT_DIRECTIVE`, `QUERY_DIRECTIVE`, `NOTE_DIRECTIVE`, `DOCUMENT_DIRECTIVE`, `PRICE_DIRECTIVE`, `COMMODITY_DIRECTIVE`, `PUSHTAG_DIRECTIVE`, `POPTAG_DIRECTIVE`, `PUSHMETA_DIRECTIVE`, `POPMETA_DIRECTIVE`. The generic `DIRECTIVE` umbrella kind from phase 2.0 is retained for error-recovery wrappers and shape-uniform tests.
- **`parse_structured(source) -> SyntaxNode`** — streaming structural parser over the lossless token stream. Recognizes each top-level directive shape (DATE+keyword for the 10 dated ones; standalone keyword for the 4 push/pop ones), opens an appropriate `*_DIRECTIVE` node, applies the Directive-Terminator Rule (trivia attaches per phase 2.0's policy), and closes at the terminator NEWLINE. Unrecognized lines (TRANSACTION, OPTION, INCLUDE, PLUGIN, CUSTOM) flow through as flat `SOURCE_FILE` children — pass-through bytes preserved, no node wrapping until later PRs cover them.
- **27 source-driven integration tests** at `crates/rustledger-parser/tests/structured_directives.rs`: one per directive type asserting exact child sequences, plus 6 trivia-policy tests verifying the Directive-Terminator Rule corollaries on real Beancount source, plus pass-through tests for TRANSACTION/OPTION/mixed files.
- **Corpus baseline switched to `parse_structured`** at `tests/baselines/cst-corpus.manifest`. The `token_seq_blake3` column is byte-identical to phase 2.0 (no token-classification change — the contract from the manifest split holds). The `node_shape_blake3` column churns for every corpus file containing any recognized directive (~half the corpus). 714 entries pass strict-mode.
- **`every_kind_partitions_token_xor_node` and `nodes_are_not_tokens`** updated with all 14 new node kinds.

## What this PR does NOT add

- **TRANSACTION header recognition** — PR 2.1b ships this paired with PR 2.2's posting body parsing.
- **Edge directives** (OPTION, INCLUDE, PLUGIN, CUSTOM) — PR 2.3.
- **Per-directive metadata sub-lines** (indented `key: "value"` lines after a directive) — left flat for a future PR that introduces `META_ENTRY` structure.
- **Typed-AST surface** (e.g., `Open::account()` accessors) — phase 3.1.

## The Directive-Terminator Rule, applied

Each recognized directive node spans content tokens + same-line trailing trivia + terminator NEWLINE per phase 2.0's [Directive-Terminator Rule](https://github.com/rustledger/rustledger/blob/main/crates/rustledger-parser/src/cst/trivia.rs). Source-driven tests cover all five corollaries:

| Test | Pins |
|---|---|
| `rule_1_same_line_trailing_comment_inside_directive` | Same-line trailing trivia stays inside |
| `rule_2_blank_line_leads_following_directive` | Blank line leads next directive |
| `rule_3_copyright_header_under_source_file` | File-leading trivia → SOURCE_FILE |
| `rule_4_trailing_comment_block_under_source_file` | File-trailing trivia → SOURCE_FILE |
| `rule_5_unterminated_final_directive` | No terminator = directive ends at last content |
| `rule_5_unterminated_with_same_line_trailing_trivia` | Rules 1+5 combo (no NL but trailing trivia) |

## Verified

- 171 unit + 4 + 1 + 1 + 55 + **27 new structured-directive tests** + 2 doctests pass.
- `cargo check --workspace --all-features --all-targets` clean.
- `cargo clippy -p rustledger-parser --tests -- -D warnings` clean.
- `cargo fmt --all --check` clean.
- `cargo deny check bans` clean.
- `RUSTDOCFLAGS=-Dwarnings cargo doc -p rustledger-parser --no-deps` clean.
- `STRICT_BASELINE=1 cargo test -p rustledger-parser --test cst_baseline` passes against the regenerated 714-file manifest.

## Test plan

- [x] One directive-shape test per kind (14 directive tests).
- [x] Trivia policy corollaries (6 tests).
- [x] Pass-through for unrecognized content (TRANSACTION, OPTION, mixed).
- [x] Edge cases (empty file, only trivia, BOM, multiple directives).
- [x] Corpus baseline: token sequence stable (contract); node shape diverges per-file (expected).

Refs #1262.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/rustledger-parser/src/cst/parser.rs [hunk 1]
```diff
@@ -25,14 +44,166 @@ pub fn parse_flat(source: &str) -> SyntaxNode {
...
+        SyntaxKind::PUSHMETA_KW => Some(SyntaxKind::PUSHMETA_DIRECTIVE),
+        SyntaxKind::POPMETA_KW => Some(SyntaxKind::POPMETA_DIRECTIVE),
+
+        // Dated directives — peek past trivia for the keyword.
+        SyntaxKind::DATE => {
+            let mut j = i + 1;
+            while j < tokens.len() && tokens[j].0.is_trivia() {
+                j += 1;
+            }
+            let (next, _) = tokens.get(j)?;
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-parser/src/cst/parser.rs]: This was already fixed in commit 01085959 (v2 of this PR). `identify_single_line_directive` now only skips `SyntaxKind::WHITESPACE` between DATE and the keyword — NEWLINE, COMMENT, and other trivia stop the walk. Pinned by the regression test `malformed_date_then_keyword_on_next_line_is_not_a_directive`. Copilot reviewed an earlier commit on this branch.

File: crates/rustledger-parser/src/cst/parser.rs [hunk 2]
```diff
@@ -25,14 +44,166 @@ pub fn parse_flat(source: &str) -> SyntaxNode {
...
+        }
+
+        // Non-trivia at the top level. Identify what kind of line
+        // starts here.
+        if let Some(directive_kind) = identify_single_line_directive(&tokens, i) {
+            // Per the Directive-Terminator Rule, pending trivia is
+            // FILE-LEADING (SOURCE_FILE direct child) if this is the
+            // first directive we've recognized; otherwise it's
+            // LEADING of this new directive (children of the
+            // directive node).
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-parser/src/cst/parser.rs]: Real documentation accuracy issue — fixed in the next commit. The flag tracks 'first non-trivia content of any kind' (recognized OR unrecognized line); the inline comment was inaccurate about the trigger. Updated to: 'pending trivia is FILE-LEADING only when NO non-trivia content has appeared yet anywhere in the file — `seen_first_content` tracks that and flips on the FIRST non-trivia content, recognized or unrecognized.' The behavior itself is correct per Roslyn (any trivia after a content line associates with the following content).
